core/modules/replica/HttpProcessor.cc

```diff
@@ -168,217 +430,835 @@ void HttpProcessor::_getReplicationLevel(qhttp::Request::Ptr req,
-                unsigned int const level = entry.first;
-                size_t const numChunks   = entry.second;
-                double const percent     = 100. * numChunks / numOnlineReplicationSystemChunks;
-                databaseJson["levels"][level]["replication"]["online"]["num_chunks"] = numChunks;
-                databaseJson["levels"][level]["replication"]["online"]["percent"   ] = percent;
+            auto itr = delays.find(worker);
+            if (delays.end() != itr) {
+                workerJson["replication"]["probe_delay_s"] = itr->second["replication"];
+                workerJson["qserv"      ]["probe_delay_s"] = itr->second["qserv"];
+            } else {
+                workerJson["replication"]["probe_delay_s"] = 0;
+                workerJson["qserv"      ]["probe_delay_s"] = 0;
             }
-            for (auto&& entry: allReplicationSystemLevels) {
-                unsigned int const level = entry.first;
-                size_t const numChunks   = entry.second;
-                double const percent     = 100. * numChunks / numAllReplicationSystemChunks;
-                databaseJson["levels"][level]["replication"]["all"]["num_chunks"] = numChunks;
-                databaseJson["levels"][level]["replication"]["all"]["percent"   ] = percent;
-            }
-            {
-                double const percent = 100. * numOrphanQservChunks / numAllQservChunks;
-                databaseJson["levels"][0]["qserv"]["online"]["num_chunks"] = numOrphanQservChunks;
-                databaseJson["levels"][0]["qserv"]["online"]["percent"   ] = percent;
-            }
-            {
-                double const percent     = 100. * numOrphanReplicationSystemChunks / numAllReplicationSystemChunks;
-                databaseJson["levels"][0]["replication"]["online"]["num_chunks"] = numOrphanReplicationSystemChunks;
-                databaseJson["levels"][0]["replication"]["online"]["percent"   ] = percent;
-            }
-            resultJson["families"][family]["databases"][database] = databaseJson;
+            resultJson.push_back(workerJson);
         }
+        resp->send(resultJson.dump(), "application/json");
+
+    } catch (exception const& ex) {
+        _error(string(__func__) + " operation failed due to: " + string(ex.what()));
+        resp->sendStatus(500);
     }
+}
+
+
+void HttpProcessor::_getWorkerStatus(qhttp::Request::Ptr req,
+                                     qhttp::Response::Ptr resp) {
+    _debug(__func__);
+    resp->sendStatus(404);
+}
+
+
+void HttpProcessor::_listControllers(qhttp::Request::Ptr req,
+                                     qhttp::Response::Ptr resp) {
+    _debug(__func__);
     
-    // Update the cache
-    _replicationLevelReport = resultJson.dump();
-    _replicationLevelReportTimeMs = PerformanceUtils::now();
+    try {
+
+        // Extract optional parameters of the query
+
+        uint64_t const fromTimeStamp = ::getQueryParam(req->query, "from");
```

Comment:
I think the `::` tells the compiler to look the function up in the global namespace, is that right? As far as I can tell these functions are declared in this file in the anonymous namespace. Do you need the `::`? I'd like to understand this better.

Reply:
That's an interesting problem. Formally, symbols defined in the unnamed namespaces appear to a code as if they're global (`extern` is the real Evil in C++. AFAIK XRootD is the only code which still uses them in `qserv`), even though (technically) they aren't the same. That's why the compiler is happy with what I'm doing above. I'm also using prefix `::` to indicate that the symbol isn't defined in a scope of a class (or any of its base classes).